src/plugins/gimbal/include/plugins/gimbal/gimbal_device.h

```diff
@@ -0,0 +1,177 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+
+#include "plugin_base.h"
+
+namespace mavsdk {
+
+class GimbalDeviceImpl;
+class System;
+
+/**
+ * @brief the GimbalDevice class provides callbacks for listening to gimbal device
+ * commands. It also provides the ability to command a gimbal device, i. e. from
+ * a custom gimbal manager.
+ *
+ * Warning: This is used for low level gimbal device control. this API should
+ * only be used if you are 1) testing a gimbal manager or 2) writing a
+ * custom gimbal manager, i. e. for a ground station.
+ */
+class GimbalDevice : public PluginBase {
+public:
+    /**
+     * @brief Constructor. Creates the plugin for a specific System.
+     *
+     * The plugin is typically created as shown below:
+     *
+     *      ```cpp
+     *      auto gimbal_device = std::make_shared<GimbalDevice>(system);
+     *      ```
+     *
+     * @param system The specific system associated with this plugin.
+     */
+    explicit GimbalDevice(System& system);
+
+    /**
+     * @brief Destructor (internal use only).
+     */
+    ~GimbalDevice();
+
+    /**
+     * @brief Gimbal device capabilities.
+     */
+    enum class Capabilities {
+        HAS_RETRACT = 1, /**< @brief Supports retracted mode. */
+        HAS_NEUTRAL = 2, /**< @brief Supports neutral mode. */
+        HAS_ROLL_AXIS = 4, /**< @brief Supports roll axis control. */
+        HAS_ROLL_FOLLOW = 8, /**< @brief Supports follow mode for roll axis. */
+        HAS_ROLL_LOCK = 16, /**< @brief Supports lock mode for roll axis. */
+        HAS_PITCH_AXIS = 32, /**< @brief Supports pitch axis control. */
+        HAS_PITCH_FOLLOW = 64, /**< @brief Supports follow mode for pitch axis. */
+        HAS_PITCH_LOCK = 128, /**< @brief Supports lock mode for pitch axis. */
+        HAS_YAW_AXIS = 256, /**< @brief Supports yaw axis control. */
+        HAS_YAW_FOLLOW = 512, /**< @brief Supports follow mode for yaw axis. */
+        HAS_YAW_LOCK = 1024, /**< @brief Supports lock mode for yaw axis. */
+        HAS_INFINITE_YAW = 2048 /**< @brief Supports infinite yaw. */
```

Comment:
I wonder if we should duplicate this, or if maybe the gimbal device tester could be implemented using an example which uses the passthrough plugin and uses just direct MAVLink.